src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -154,217 +162,180 @@ def process(
-                cc.warning(
-                    "Discovered extension functions that have value None or do not contain any information."
-                )
-                cc.warning(
-                    "Skipping generating code of all such objects from this file."
+                cc.message(
+                    "-- Generating Snowpark annotation SQL code for {}".format(py_file)
                 )
-
-            # 4. Enrich the raw extension functions by setting additional properties
-            for raw_ex_fn in filtered_collection:
-                sanitize_extension_function_data(ex_fn=raw_ex_fn, py_file=dest_file)
-                enrich_ex_fn(
-                    ex_fn=raw_ex_fn,
-                    py_file=dest_file,
-                    deploy_root=self.deploy_root,
+                cc.message(create_stmt)
+                collected_output.append(
+                    f"-- {py_file.relative_to(bundle_map.deploy_root())}"
                 )
+                collected_output.append(create_stmt)
 
-            dest_file_py_file_to_collected_raw_ex_fns[dest_file] = filtered_collection
+                grant_statements = generate_grant_sql_ddl_statements(extension_fn)
+                if grant_statements is not None:
+                    cc.message(grant_statements)
+                    collected_output.append(grant_statements)
 
-        # For each extension function, generate its related SQL statements
-        dest_file_py_file_to_ddl_map: Dict[
-            Path, str
-        ] = self.generate_sql_ddl_statements(dest_file_py_file_to_collected_raw_ex_fns)
+        return "\n".join(collected_output)
 
-        # TODO: Temporary for testing, while feature is being built in phases
-        return dest_file_py_file_to_ddl_map
+    def _normalize(self, extension_fn: NativeAppExtensionFunction, py_file: Path):
+        if extension_fn.name is None:
+            # The extension function was not named explicitly, use the name of the Python function object as its name
+            extension_fn.name = extension_fn.handler
 
-    def generate_sql_ddl_statements(
-        self, dest_file_py_file_to_collected_raw_ex_fns: Dict[Path, Optional[Any]]
-    ) -> Dict[Path, str]:
-        """
-        Generates SQL DDL statements based on the entities collected from a set of python files in the artifact_to_process.
-        """
-        dest_file_py_file_to_ddl_map: Dict[Path, str] = {}
-        for py_file in dest_file_py_file_to_collected_raw_ex_fns:
+        # Compute the fully qualified handler
+        extension_fn.handler = f"{py_file.stem}.{extension_fn.handler}"
 
-            collected_ex_fns = dest_file_py_file_to_collected_raw_ex_fns[
-                py_file
-            ]  # Collected entities is List[Dict[str, Any]]
-            if collected_ex_fns is None:
-                continue
+        if extension_fn.imports is None:
+            extension_fn.imports = []
+        extension_fn.imports.append(f"/{to_stage_path(py_file)}")
```

Comment:
1. `NativeAppExtensionFunction` does not have an `imports` attribute.
2. What if the user has already specified the correct path as part of the decorator? This will duplicate that, unless we make it a set.

Reply:
1. Yes it does, inherited from the Snowpark model
2. Yup, we'll duplicate it for now. Not the final version. We need to rehash how we handle imports anyway. Just focusing on enabling e2e testing in here.

Reply:
1. ack
2. I see, can be part of the PR I'm working on.